extjsdk/src/main/java/io/vantiq/extjsdk/ExtensionWebSocketClient.java

```diff
@@ -180,6 +181,12 @@ public ExtensionWebSocketClient (String sourceName) {
         this(sourceName, DEFAULT_FAILED_MESSAGE_QUEUE_SIZE);
     }
 
+    /**
+     * Unique identifier for this loader's class.  This is set on load and then left.  This
+     * shared secret allows instances of this client to perform reconnects to the server.
+     */
+    protected final static String clientReconnectSecret = UUID.randomUUID().toString();
+
```

Comment:
This is the actual secret.  It is generated at class load/initialization time and allows this connector instance to reconnect at will. It will be the same for all sources to which this connector connects, but that's irrelevant -- we just want to prove ownership.